bertopic/plotting/_datamap.py

```diff
@@ -0,0 +1,134 @@
+    docs = fetch_20newsgroups(subset='all',  remove=('headers', 'footers', 'quotes'))['data']
+    sentence_model = SentenceTransformer("all-MiniLM-L6-v2")
+    embeddings = sentence_model.encode(docs, show_progress_bar=False)
+
+    # Train BERTopic
+    topic_model = BERTopic().fit(docs, embeddings)
+
+    # Reduce dimensionality of embeddings, this step is optional
+    # reduced_embeddings = UMAP(n_neighbors=10, n_components=2, min_dist=0.0, metric='cosine').fit_transform(embeddings)
+
+    # Run the visualization with the original embeddings
+    topic_model.visualize_document_datamap(docs, embeddings=embeddings)
+
+    # Or, if you have reduced the original embeddings already:
+    topic_model.visualize_document_datamap(docs, reduced_embeddings=reduced_embeddings)
+    ```
+
+    Or if you want to save the resulting figure:
+
+    ```python
+    fig = topic_model.visualize_document_datamap(docs, reduced_embeddings=reduced_embeddings)
+    fig.savefig("path/to/file.png", bbox_inches="tight")
+    ```
+    <img src="../../getting_started/visualization/datamapplot.png",
+         alt="DataMapPlot of 20-Newsgroups", width=800, height=800></img>
+    """
+
+    topic_per_doc = topic_model.topics_
+
+    df = pd.DataFrame({"topic": np.array(topic_per_doc)})
+    df["doc"] = docs
+    df["topic"] = topic_per_doc
+
+    # Extract embeddings if not already done
+    if embeddings is None and reduced_embeddings is None:
+        embeddings_to_reduce = topic_model._extract_embeddings(df.doc.to_list(), method="document")
+    else:
+        embeddings_to_reduce = embeddings
+
+    # Reduce input embeddings
+    if reduced_embeddings is None:
+        umap_model = UMAP(n_neighbors=15, n_components=2, min_dist=0.15, metric='cosine').fit(embeddings_to_reduce)
+        embeddings_2d = umap_model.embedding_
+    else:
+        embeddings_2d = reduced_embeddings
+
+    unique_topics = set(topic_per_doc)
+
+    # Prepare text and names
+    if isinstance(custom_labels, str):
+        names = [[[str(topic), None]] + topic_model.topic_aspects_[custom_labels][topic] for topic in unique_topics]
+        names = [" ".join([label[0] for label in labels[:4]]) for labels in names]
+        names = [label if len(label) < 30 else label[:27] + "..." for label in names]
+    elif topic_model.custom_labels_ is not None and custom_labels:
+        names = [topic_model.custom_labels_[topic + topic_model._outliers] for topic in unique_topics]
+    else:
+        names = [f"Topic-{topic}: " + " ".join([word for word, value in topic_model.get_topic(topic)][:3]) for topic in unique_topics]
+
+    topic_name_mapping = {topic_num: topic_name for topic_num, topic_name in zip(unique_topics, names)}
+    topic_name_mapping[-1] = "Unlabelled"
+    named_topic_per_doc = pd.Series(topic_per_doc).map(topic_name_mapping).values
+
+    figure, axes = datamapplot.create_plot(
+        embeddings_2d,
+        named_topic_per_doc,
+        figsize=(width/100, height/100),
+        dpi=100,
```

Comment:
Out of curiosity, why did you set this at a default of 100? Is the default `dpi` not sufficient for these plots?

Reply:
100 is the default DPI for matplotlib. To keep with "width x height" in pixels as input params (following the other plotting methods) it helps if the DPI stays fixed, so I set it to the default.